inc/saiuburst.h

```diff
@@ -0,0 +1,401 @@
+ * @param[in] switch_id Switch object id
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_create_tam_histogram_fn)(
+        _Out_ sai_object_id_t *tam_histogram_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Delete a specified histogram object and free driver memory
+ *
+ * @param[in] tam_histogram_id Histogram object to be removed.
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_remove_tam_histogram_fn)(
+        _In_ sai_object_id_t tam_histogram_id);
+
+/**
+ * @brief Set histogram attribute value(s)
+ *
+ * @param[in] tam_histogram_id Histogram object id
+ * @param[in] attr Attribute to set
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_set_tam_histogram_attribute_fn)(
+        _In_ sai_object_id_t tam_histogram_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get values for specified histogram attributes.
+ *
+ * @param[in] tam_histogram_id Histogram object id
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_tam_histogram_attribute_fn)(
+        _In_ sai_object_id_t tam_histogram_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Obtain the values for all bins from a histogram.
+ *
+ * Values array must supply sufficient memory for values of all
+ * bins as specified for the histogram object.
+ *
+ * @param[in] tam_histogram_id Histogram object id
+ * @param[inout] number_of_counters Number of bins (required/provided)
+ * @param[out] counters Statistics values (allocated/provided)
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_tam_histogram_stats_fn)(
+        _In_ sai_object_id_t tam_histogram_id,
+        _Inout_ uint32_t *number_of_counters,
+        _Out_ uint64_t *counters);
+
+typedef struct _sai_uburst_api_t
+{
+    sai_create_tam_microburst_fn            create_tam_microburst;
+    sai_remove_tam_microburst_fn            remove_tam_microburst;
+    sai_set_tam_microburst_attribute_fn     set_tam_microburst_attribute;
+    sai_get_tam_microburst_attribute_fn     get_tam_microburst_attribute;
+    sai_create_tam_histogram_fn             create_tam_histogram;
+    sai_remove_tam_histogram_fn             remove_tam_histogram;
+    sai_set_tam_histogram_attribute_fn      set_tam_histogram_attribute;
+    sai_get_tam_histogram_attribute_fn      get_tam_histogram_attribute;
+    sai_get_tam_histogram_stats_fn          get_tam_histogram_stats;
```

Comment:
if stats can be cleared, please also add  "sai_clear_tam_histogram_stats_fn" function